sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkSessionOperation.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Analytics.Synapse.Spark.Models;
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// A long-running operation for:
+    /// <see cref="SparkSessionClient.StartCreateSparkSession(SparkSessionOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkSessionClient.StartCreateSparkSessionAsync(SparkSessionOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkSessionClient.StartGetSparkSession(int, bool?, CancellationToken)"/>,
+    /// <see cref="SparkSessionClient.StartGetSparkSessionAsync(int, bool?, CancellationToken)"/>
+    /// </summary>
+    public class SparkSessionOperation : Operation<SparkSession>
+    {
+        private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(5);
+
+        private readonly ClientDiagnostics _diagnostics;
+
+        private readonly SparkSessionClient _client;
+        private readonly SparkSession _value;
+        private Response<SparkSession> _response;
+        private bool _completed;
+
+        internal SparkSessionOperation(SparkSessionClient client, ClientDiagnostics diagnostics, Response<SparkSession> response)
+        {
+            _client = client;
+            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
```

Comment:
Can this happen?

Reply:
I don't think so, but the code I modeled this off of did it this way.